### PR TITLE
Migrate logging to logrus

### PR DESCRIPTION
**Pull request by gabrieljackson**
This standardizes logging with the rest of the cloud infrastructure and allowed for easy transition off of the Mattermost server v5 dependency. Matterwick logic should be unchanged, but logging fields should be better standardized.

Note that this is just a first pass and there is more work we can do to improve logging further.

https://mattermost.atlassian.net/browse/MM-52148

```release-note
Migrate logging to logrus
```

### CODE REVIEW

File: server/logger.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+func init() {
+	logger = log.New()
+	logger.SetFormatter(&log.TextFormatter{
+		FullTimestamp: true,
+	})
+	// Output to stdout instead of the default stderr.
+	log.SetOutput(os.Stdout)
+}
+
+type logrusWriter struct {
```
Comment (mirshahriar) [inline server/logger.go:24]: Do we need this? I can't see any usage.

Comment (mirshahriar) [inline server/logger.go:24]: Do we need this? I don't see any usage.

Comment (gabrieljackson) [reply server/logger.go:24]: Yep, good catch. That was a copy-pasta that we don't seem to need.

**Review by mirshahriar — APPROVED** [review]
LGTM

**Pull request merged**